src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDb.java

```diff
@@ -95,4 +97,33 @@ public String toString() {
         }
     }
 
+    /**
+     * Wrap a scan result from a multi tenant scan, to associate tenant information per scan result item.
+     **/
+    class MtScanResult extends ScanResult {
```

Comment:
not crazy about how this is done, which is fragile if ScanResult gets changed in a newer aws version, open to feedback

Reply:
Ideally, we'd want tenant and table name attached to each time in the scan result, right? Have you considered putting items with special keys into the map, such as `mt:contxt` and `mt:tableName`?